pkg/runtime/runtime.go

```diff
@@ -516,13 +519,26 @@ func (a *DaprRuntime) beginPubSub(name string, ps pubsub.PubSub) error {
 				}
 			}
 
-			if pubsub.HasExpired(cloudEvent) {
-				log.Warnf("dropping expired pub/sub event %v as of %v", cloudEvent[pubsub.IDField], cloudEvent[pubsub.ExpirationField])
+			route := a.topicRoutes[msg.Metadata[pubsubName]].routes[msg.Topic]
+			isdlq, _ := strconv.ParseBool(route.metadata["IsDLQ"])
 
-				return nil
+			if pubsub.HasExpired(cloudEvent) && !isdlq {
+				log.Warnf("dropping expired pub/sub event %v as of %v", cloudEvent[pubsub.IDField], cloudEvent[pubsub.ExpirationField])
+				if route.metadata["DLQTopic"] != "" && route.metadata["DLQPubsubName"] != "" {
+					log.Warnf("pub/sub message %v being sent to dlq topic %v", cloudEvent[pubsub.IDField], route.metadata["DLQTopic"])
+					if err := a.Publish(&pubsub.PublishRequest{
+						Data:       msg.Data,
+						PubsubName: route.metadata["DLQPubsubName"],
+						Topic:      route.metadata["DLQTopic"],
+						Metadata:   msg.Metadata,
+					}); err != nil {
+						log.Warnf("expired message %v failed to go to DLQ", cloudEvent[pubsub.IDField])
```

Comment:
You can log error in this case since it is pretty serious that the message was given up. I think we might want to add the retry logic here.

/cc @pkedy